libraries/rush-lib/src/logic/operations/LogChunksWritable.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { TerminalWritable, type ITerminalChunk } from '@rushstack/terminal';
+
+import type { RushConfigurationProject } from '../../api/RushConfigurationProject';
+import { RushConstants } from '../RushConstants';
+import { getRelativeLogFilePathBase } from './ProjectLogWritable';
+
+import Stringer from 'stream-json/jsonl/Stringer';
+import { chain } from 'stream-chain';
```

Comment:
Although it's only 2 NPM packages, this is 126k of extra logic being pulled into Rush. Is it really necessary to accomplish what seems like `JSON.serialize(x).join('\n')` but with buffering? 🤔 (I don't know the answer, just curious.)

Reply:
Good call, I was initially cautious of stream interruptions using JSON along so I ended up going for a streaming JSON lines approach, but we should be able to just do this in memory. I'll adjust.